### ISSUE TITLE
filename ordering / exception mutation breaks `exception-message-attribute`

### ISSUE THREAD

**Issue by asottile** [PyCQA/pylint#2893]
### Steps to reproduce

I have the following minimal setup:

```python
# a.py

def f():
    e = Exception()
    e.message = 'bang!'
    raise e
```

```python
# b.py

def g():
    try:
        raise TypeError('ohai')
    except Exception as e:
        print(e.message)
```

### Current behavior

Currently, errors are reported based on the *ordering* of the invocation:

```console
$ pylint --py3k a.py b.py

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

```console
$ pylint --py3k b.py a.py
************* Module b
b.py:7:14: W1645: Exception.message removed in Python 3 (exception-message-attribute)

------------------------------------------------------------------
Your code has been rated at 8.89/10 (previous run: 8.89/10, +0.00)
```

### Expected behavior

I expect both to produce the error from `b.py`, and maybe even an error in `a.py` for assigning to `Exception.message`

### pylint --version output

```
$ pylint --version
pylint 2.3.1
astroid 2.2.5
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0]
```

**Comment by PCManticore** [comment]
Thanks for the report! This looks like https://github.com/PyCQA/pylint/issues/689 but we should definitely emit the same error for both files, regardless from where the attribute came.
